lib/ecto_job/worker.ex

```diff
@@ -17,12 +18,18 @@ defmodule EctoJob.Worker do
   Start a worker process given a repo module and a job struct
   This may fail if the job reservation has expired, in which case the job will be
   reactivated by the producer.
+  SimpleDemo.JobQueue.enqueue("jim")
   """
   @spec start_link(repo, EctoJob.JobQueue.job(), DateTime.t()) :: {:ok, pid}
   def start_link(repo, job = %queue{}, now) do
     Task.start_link(fn ->
       with {:ok, job} <- JobQueue.update_job_in_progress(repo, job, now) do
-        queue.perform(JobQueue.initial_multi(job), job.params)
+        case queue.perform(JobQueue.initial_multi(job), job.params) do
+          {:ok, res} -> IO.inspect(res)
+          {:error, reason} -> JobQueue.update_error(repo, job, reason)
+          {:error, _ , message, job} -> JobQueue.update_error(repo, extract_job(job), message)
+          error -> IO.inspect(error)
```

Comment:
If the return value doesn't match `{:ok, res}` or `{:error, reason}` or `{:error, failed_operation, failed_value, changes_so_far}` then this code should probably crash, since the user has made a mistake.